Fabric.Authorization.Domain/Services/EDWAdminRoleSyncService.cs

```diff
@@ -0,0 +1,71 @@
+﻿namespace Fabric.Authorization.Domain.Services
+{
+    using System;
+    using System.Collections.Generic;
+    using System.Linq;
+    using System.Threading.Tasks;
+    using Fabric.Authorization.Domain.Models;
+    using Fabric.Authorization.Domain.Models.EDW;
+    using Fabric.Authorization.Domain.Stores;
+
+    public class EDWAdminRoleSyncService
+    {
+        private readonly RoleService _roleService;
+        private readonly GroupService _groupService;
+        private readonly IEDWStore _edwStore;
+
+
+        public EDWAdminRoleSyncService(RoleService roleService, GroupService groupService, IEDWStore edwStore)
+        {
+            this._roleService = roleService ??
+                    throw new ArgumentNullException(nameof(roleService));
+            this._groupService = groupService ??
+                    throw new ArgumentNullException(nameof(groupService));
+            this._edwStore = edwStore ??
+                    throw new ArgumentNullException(nameof(edwStore));
+        }
+
+        public async Task RefreshDosAdminRolesAsync(User user)
+        {
+            if (await this.IsUserASuperAdmin(user))
+            {
+                _edwStore.AddIdentitiesToRole(new[] { user.SubjectId }, EDWConstants.EDWAdmin);
+            }
+            else
+            {
+                _edwStore.RemoveIdentitiesFromRole(new[] { user.SubjectId }, EDWConstants.EDWAdmin);
+            }
+        }
+
+        public async Task RefreshDosAdminRolesAsync(IEnumerable<User> users)
+        {
+            foreach(User user in users)
```

Comment:
Looks like we are hitting the database on each loop - is there a more efficient way to do this?

Reply:
We started looking at the database calls to make sure we need to make them.

Reply:
Hey @michaelvidal24 , we have added feature flag and are ready to complete this pull request